hugegraph-hubble/hubble-dist/assembly/travis/run_live_hubble_smoke.py

```
@@ -0,0 +1,638 @@
+            response = request("GET", f"{hubble_url}/actuator/health", timeout=3)
+            if isinstance(response, dict) and response.get("status") == "UP":
+                return response
+        except Exception as exc:  # noqa: BLE001 - smoke tool should report final error
+            last_error = exc
+        time.sleep(1)
+    raise RuntimeError(f"Hubble health did not become UP: {last_error}")
+
+
+def extract_tarball(tarball, work_dir):
+    with tarfile.open(tarball, "r:gz") as archive:
+        archive.extractall(work_dir)
```

Ruff (0.15.17)

[error] Uses of `tarfile.extractall()`

(S202)